fair/src/autogluon/fair/utils/group_metrics.py

```diff
@@ -0,0 +1,240 @@
+neg_pred_val = GroupMetric(lambda TP, FP, FN, TN: (TN) / (1e-6 + TN + FN), 'Negative Predicted Value')
+
+# Existing binary metrics for autogluon
+accuracy = GroupMetric(lambda TP, FP, FN, TN: (TP + TN) / (TP + FP + FN + TN), 'Accuracy')
+balanced_accuracy = GroupMetric(lambda TP, FP, FN, TN: (TP / (1e-6 + TP + FN) + TN / (1e-6 + TN + FP)) / 2,
+                                'Balanced Accuracy')
+min_label_accuracy = GroupMetric(lambda TP, FP, FN, TN: np.minimum(TP / (1e-6 + TP + FN), TN / (1e-6 + TN + FP)),
+                                 'Minimum-Label-Accuracy')#common in min-max fairness literature
+f1 = GroupMetric(lambda TP, FP, FN, TN: (2 * TP) / (1e-6 + 2 * TP + FP + FN), 'F1 score')
+precision = pos_pred_val
+recall = true_pos_rate
+mcc = GroupMetric(lambda TP, FP, FN, TN:
+                  (TP * TN - FP * FN) / (1e-6 + np.sqrt((TP + FP) * (TP + FN) * (TN + FP) * (TN + FN))),
+                  'MCC')
+
+ag_metrics = {'accuracy': accuracy,
+              'balanced_accuracy': balanced_accuracy,
+              'f1': f1,
+              'mcc': mcc,
+              'precision': precision,
+              'recall': recall,
+              'roc_auc': roc_auc}
+count_metrics = {'pos_data_count': pos_data_count,
+                 'neg_data_count': neg_data_count,
+                 'pos_data_rate': pos_data_rate,
+                 'pos_pred_rate': pos_pred_rate}
+default_group_metrics = ag_metrics | count_metrics
+
+# Postprocessing Clarify metrics  https://mkai.org/learn-how-amazon-sagemaker-clarify-helps-detect-bias
+class_imbalance = pos_data_rate.diff
+demographic_parity = pos_pred_rate.diff
+disparate_impact = pos_pred_rate.ratio
+acceptance_rate = precision
+cond_accept = GroupMetric(lambda TP, FP, FN, TN: (TP + FN) / (1e-6 + TP + FP), 'Conditional Acceptance Rate')
+cond_reject = GroupMetric(lambda TP, FP, FN, TN: (TN + FP) / (1e-6 + TN + FN), 'Conditional Rejectance Rate')
+specificity = true_neg_rate
+rejection_rate = neg_pred_val
+error_ratio = GroupMetric(lambda TP, FP, FN, TN: FP / (1e-6 + FN), 'Error Ratio')
+treatment_equality = error_ratio.diff
+clarify_metrics = {
+    'class_imbalance': class_imbalance,
+    'demographic_parity': demographic_parity,
+    'disparate_impact': disparate_impact,
+    'cond_accept.diff': cond_accept.diff,
+    'cond_reject.diff': cond_reject.diff,
+    'specificity.diff': specificity.diff,
+    'recall.diff': recall.diff,
+    'acceptance_rate.diff': acceptance_rate.diff,
+    'rejection_rate.diff': rejection_rate.diff,
+    'accuracy.diff': accuracy.diff,
+    'treatment_equality': treatment_equality}
+
+# Existing fairness definitions. Binary definitions from: https://fairware.cs.umass.edu/papers/Verma.pdf
+stastical_parity = demographic_parity
+predictive_parity = precision.diff
+neg_predictive_parity = neg_pred_val.diff
+predictive_equality = false_neg_rate.diff
+equal_opportunity = recall.diff
+
+
+class AddGroupMetrics(BaseGroupMetric):
```

Comment:
Why stop at 2 metrics? Seems like an unnecessary limitation.

Reply:
There's two fairness metrics I could find in the literature that are the sum of two other metrics.  Equalized odds and conditional use accuracy. But I don't know any that sum together three.

Happy to move to the general case if you think it's worthwhile.